### PR TITLE
[#4609] nuffic reports

### LINKED ISSUE #4609: Nuffic reports
- Project level report, add pending approval tabs. One sheet for approved information
- Flat programme export excel table
- Activate automatic aggregation of target values

### PR DESCRIPTION
**Pull request by zuhdil**
- [ ] Test plan | Unit test | Integration test
- [ ] Documentation

Closes #4609

### CODE REVIEW

File: akvo/rsr/dataclasses.py [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+
+def group_results_by_types(results):
+    types = {}
+    for result in results:
+        type = result.iati_type_name
+        if not type:
+            continue
+        if type not in types:
+            types[type] = []
+        types[type].append(result)
```
Comment (MichaelAkvo) [inline akvo/rsr/dataclasses.py:517]: Could use `dict.setdefault()` here.

File: akvo/rsr/views/py_reports/utils.py [hunk 2]
```diff
@@ -102,6 +102,12 @@ def get_order_or_id_attribute(item):
     return item.order + 1 if item.order is not None else item.id
 
 
+def is_using_indicator_target(project):
+    program = project.get_program()
+    targets_at = program.targets_at if program else project.targets_at
+    return True if targets_at == 'indicator' else False
```
Comment (MichaelAkvo) [inline akvo/rsr/views/py_reports/utils.py:108]: No need for a ternary

**Review by MichaelAkvo — CHANGES_REQUESTED** [review]
I probably missed something since the PR is so big, but here are a few things to change or consider.

File: akvo/rsr/dataclasses.py [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+            return calculate_percentage(ensure_decimal(item.numerator), ensure_decimal(item.denominator))
+        return item.value
+
+    def get_aggregated_disaggregation_value(self, category, type):
+        item = self._select_disaggregation(self.aggregated_disaggregations, category, type)
+        if not item:
+            return None
+        if self.is_percentage:
+            return calculate_percentage(ensure_decimal(item.numerator), ensure_decimal(item.denominator))
+        return item.value
```
[the remark this answers is not in the document]
Comment (zuhdil) [reply akvo/rsr/dataclasses.py:280]: Not really, one is to get a disaggregation value of the current reporting period, and the other is to get the aggregated value of contributors (child periods) plus the current reporting period.

[the remark this answers is not in the document]
Comment (MichaelAkvo) [reply akvo/rsr/dataclasses.py:280]: Understood. I missed the `aggregated_` on line 275.

**Review by MichaelAkvo — APPROVED** [review]
LGTM

**Pull request merged**